### PR TITLE
Added batch set penalty method

### PR DESCRIPTION
**Pull request by VladasZ**

### CODE REVIEW

File: contract/src/penalty/api.rs [hunk 1]
```diff
@@ -17,13 +18,26 @@ pub trait PenaltyApi {
...
     /// # Panics
     ///
     /// This method will panic if the jar's associated product has a constant APY rather than a downgradable APY.
     fn set_penalty(&mut self, account_id: AccountId, jar_id: JarIdView, value: bool);
+
+    /// Behaves the same as `set_penalty`
+    ///
+    /// # Arguments
+    ///
+    /// * `jars` - Array containing `AccountId` and `Vec<JarIdView>` for each user and each jar.
```
Comment (vasyafromrussia) [inline contract/src/penalty/api.rs:34]: -> List of Account IDs and their Jar IDs to which penalty must be applied.

File: contract/src/penalty/api.rs [hunk 2]
```diff
@@ -45,4 +59,40 @@ impl PenaltyApi for Contract {
...
+
+                let product = self
+                    .products
+                    .get(&jar.product_id)
+                    .unwrap_or_else(|| env::panic_str(&format!("Product '{}' doesn't exist", jar.product_id)));
+
+                match product.apy {
+                    Apy::Downgradable(_) => jar.apply_penalty(value),
+                    Apy::Constant(_) => env::panic_str("Penalty is not applicable for constant APY"),
+                };
```
Comment (vasyafromrussia) [inline contract/src/penalty/api.rs:87]: The same check is used in set_penalty for a single Jar. So it's better to extract this to an assertion method

Comment (vasyafromrussia) [reply contract/src/penalty/api.rs:87]: Split it to assertion and applying of penalty, I mean

**Review by vasyafromrussia — CHANGES_REQUESTED** [review]
Small issues

File: contract/src/penalty/api.rs (hunk 1 above)
Comment (VladasZ) [reply contract/src/penalty/api.rs]: Most of my comments usually sound like a caveman speach =)

**Comment by georgebancila** [comment]
@vasyafromrussia @VladasZ hey guys 👋🏼 
Can you return `timestamp` in `PenaltyData` too? The timestamp you used when you apply the penalty, [this one](https://github.com/sweatco/sweat-jar/pull/58/files#diff-202944cbf776d1244b148fd2bb03e713332492f16423a30e3fbd545a4aae264bR68). It would help us calculate accurately the interest when showing on FE or on BE push notifications.

**Comment by VladasZ** [comment]
Done

**Review by vasyafromrussia — APPROVED** [review]

**Pull request merged**